v2/oauth/introduction.mdx

```diff
@@ -0,0 +1,138 @@
+---
+title: Introduction 
+hide_title: true
+---
+
+# OAuth2 Support 
+
+**OAuth2**, Open Authorization, is an industry-standard authorization framework that enables third-party applications to obtain limited access to a user's resources without exposing their credentials.
+
+
+## When to use OAuth2
+
+In most cases, when using **SuperTokens**, you should not have to use **OAuth2** directly.
+We already expose the **[ThirdParty Recipe](/docs/thirdparty/introduction)** that enables users to login using external providers. 
+This covers most of the authentication scenarios that involve OAuth2. 
+
+However, there are some custom use cases where you will have to implement a different authentication flow.
+
+- **[If you have multiple frontend clients that connect to different backends](/docs/oauth/multiple-frontends-with-separate-backends)**
+- **[If you have multiple frontend clients connecting to the same backend](/docs/oauth/multiple-frontends-with-a-single-backend)**
+- **[If you want to have an unified login experience across multiple independent applications](/docs/oauth/clients-without-dedicated-backend)**
+- **[If you want to reuse your website login for desktop and mobile apps](/docs/oauth/reuse-website-login)**
+- **[For authenticating microservices](/docs/oauth/machine-to-machine-authentication)**
+
+For these specific instances we expose recipes that allow you to complete your setup. 
+
+:::info
+The **OAuth2 recipes** are available only when using the **SuperTokens Managed Service**. This feature is not included in the **Self-Hosted** version.
+
+Also, the feature at the moment is supported just by the following **SDKs**:
+- `supertokens-node`
+- `supertokens-auth-react`
+
+Additionally, our **OAuth2** implementation does not support magic links based login flows right now.
+:::
+
+## OAuth2 Reference
```

Comment:
Whilst i see the value of this section, the way it is written is very hard to follow. As a reader who is not already familiar with OAuth and its terms, its really hard to actually understand the below. Im not too sure how to make it better, but if you can think of something, that would be great

Reply:
I think we should have this even if it is not useful for everybody. Some possible improvements:
- We could group them a bit better (e.g.: group all Roles together (client, provider etc), group all Flows together
- Add some kind of graphical representation for each group
- Add a list of concrete examples for every item and/or a usecase where it is relevant.

Reply:
resolved